### ISSUE TITLE
Insert into columnar table might fail due to fkeys between columnar metadata tables (v10.0.x)

### ISSUE THREAD

**Issue by SaitTalhaNisanci** [citusdata/citus#5196]
edited by @onurctirtir: Given that this is already fixed as of v10.1, the workaround is upgrading Citus to a version >= v10.1.

-----

It seems that there was an assertion failure on columnar side which is fixed in 10.1 (I will update the issue when I find the commit that fixes the issue).

This seems to be the commit that fixes the issue: https://github.com/citusdata/citus/commit/874d5fd96218d549bb573d124ace775b84c58db8

This is unrevealed when PG changed the assert to an error with this commit: ef12f32c0870b73ab7589ffc178094ac9ba3e84f

To easily reproduce, checkout to release-10.0 and:
```sql
CREATE TABLE columnar_table_1(a INT, b INT) USING columnar;
INSERT INTO columnar_table_1 SELECT i FROM generate_series(160001, 320000) i;
```

We might need to backport the fix to release-10.0

**Comment by onurctirtir** [comment]
Couldn't reproduce this on release-10.0, is there any additional steps ?

**Comment by onurctirtir** [comment]
Could only reproduce the error message on PG13.4, not the assertion failure:

```sql
-- on release-10.0
CREATE TABLE columnar_table_1(a INT, b INT) USING columnar;
INSERT INTO columnar_table_1 SELECT i FROM generate_series(160001, 320000) i;
ERROR:  cannot execute SQL without an outer snapshot or portal
CONTEXT:  SQL statement "SELECT 1 FROM ONLY "columnar"."stripe" x WHERE "storage_id" OPERATOR(pg_catalog.=) $1 AND "stripe_num" OPERATOR(pg_catalog.=) $2 FOR KEY SHARE OF x"

**Comment by SaitTalhaNisanci** [comment]
With 13.4 the assertion failure is converted to an error, that is why we hit this.

**Comment by onurctirtir** [comment]
Given that this is already fixed as of v10.1, the workaround is upgrading Citus to a version >= v10.1.
**However**, we have seen that this foreign key related error might leave the last disk block occupied by columnar table in an inconsistent state.
That is to say, assuming following scenario:
* User is on v10.0.x
* Attempts to insert into columnar table, gets the error written above^: `ERROR:  cannot execute SQL without an outer snapshot or portal`
* Upgrades citus to v10.2.x to run away from that^ error permanently.
* Attempts to insert into columnar table on v10.2.x, but this time they might get the following error: `ERROR:  attempt to write columnar data of length X to offset Y of block Z of relation T`

**In that case, user should vacuum their table to fix the last disk block of their columnar table, as in `VACUUM columnar_table;`.**

**Comment by marcocitus** [comment]
Seems fixed on recent versions
